### PR TITLE
CA-355145: guard vbd_stats

### PR DESCRIPTION
**Pull request by MarkSymsCtx**
Signed-off-by: Mark Syms <mark.syms@citrix.com>

### CODE REVIEW

**Comment by MarkSymsCtx** [comment]
https://github.com/xapi-project/blktap/pull/310 fixed some of this but was incomplete

File: drivers/td-req.c [hunk 1]
```diff
@@ -702,11 +702,13 @@ tapdisk_xenblkif_parse_request(struct td_xenblkif * const blkif,
         }
 		if (likely(blkif->stats.xenvbd))
 			blkif->stats.xenvbd->st_wr_sect += nr_sect;
-        blkif->vbd_stats.stats->write_sectors += nr_sect;
+		if (likely(blkif->vbd_stats.stats))
```
Comment (edwintorok) [inline drivers/td-req.c:705]: There are more instances of unguarded usages of 'vbd_stats.stats' in this file

Comment (edwintorok) [reply drivers/td-req.c:705]: See for example:
https://github.com/xapi-project/blktap/blob/master/drivers/td-req.c#L503
And in td-blkif.c too, try `git grep vbd_stats.stats`

Comment (edwintorok) [reply drivers/td-req.c:705]: And similarly for 'stats.xenvbd->' there are some unguarded usages in td-blkif.c right next to the vbd_stats one in tapdisk low memory mode: 
https://github.com/xapi-project/blktap/search?q=stats.xenvbd-

(unfortunately a similar search for vbd_stats.stats- yields no results on github, github bug?)

Comment (MarkSymsCtx) [reply drivers/td-req.c:705]: > There are more instances of unguarded usages of 'vbd_stats.stats' in this file

The ones in `tapdisk_xenblkif_complete_request` are guarded by the wider `if (likely(!blkif->dead))`

Comment (MarkSymsCtx) [reply drivers/td-req.c:705]: > And similarly for 'stats.xenvbd->' there are some unguarded usages in td-blkif.c right next to the vbd_stats one in tapdisk low memory mode:
> https://github.com/xapi-project/blktap/search?q=stats.xenvbd-
> 
> (unfortunately a similar search for vbd_stats.stats- yields no results on github, github bug?)

Those are in the stats creation function where it is not possible for the stats to have been freed by an asynchronous completion.

Comment (MarkSymsCtx) [reply drivers/td-req.c:705]: The ones in `tapdisk-server.c` do need looking at.

**Review by edwintorok — APPROVED** [review]

**Review by TimSmithCtx — APPROVED** [review]

**Review by rosslagerwall — APPROVED** [review]

**Pull request merged**